### PR TITLE
ENH: Export file with desired output format

### PR DESCRIPTION
**Pull request by turanoo**

### CODE REVIEW

File: q2cli/tests/test_tools.py [hunk 1]
```diff
@@ -130,6 +135,68 @@ def test_merged_metadata(self):
...
+
+    def test_export_to_dir_no_format(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.viz, '--output-path', output_path
+        ])
+
+        self.assertTrue(os.path.isdir(output_path))
+        self.assertIn('index.html', os.listdir(output_path))
+        self.assertIn('index.html', os.listdir(output_path))
```
Comment (ebolyen) [inline q2cli/tests/test_tools.py]: Drop duplicate assert

File: q2cli/tests/test_tools.py [hunk 2]
```diff
@@ -130,6 +135,68 @@ def test_merged_metadata(self):
...
+
+    def test_export_to_file(self):
+        output_path = os.path.join(self.tempdir, 'output')
+        os.mkdir(output_path)
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequence1'
+            ])
+        print(os.listdir(output_path))
```
Comment (ebolyen) [inline q2cli/tests/test_tools.py]: Looks like this print never became an assert

File: q2cli/tests/test_tools.py [hunk 3]
```diff
@@ -130,6 +135,68 @@ def test_merged_metadata(self):
...
+    #
+    #     self.runner.invoke(tools, [
+    #         'export', '--input-path', self.ints1, '--output-path', output_path,
+    #         '--output-format', 'IntSequence1'
+    #     ])
+    #     self.assertTrue(os.path.isfile(output_path))
+    #
+    #
+    # def test_export_path_dir_is_replaced(self):
+    #     assert False
```
Comment (ebolyen) [inline q2cli/tests/test_tools.py]: Tests should never be commented out. Please finish these.

File: q2cli/tests/test_tools.py [hunk 4]
```diff
@@ -130,6 +135,64 @@ def test_merged_metadata(self):
...
+
+    def test_export_path_file_is_replaced(self):
+        output_path = os.path.join(self.tempdir, 'helloWorld')
+        os.mkdir(output_path)
+
+        result = self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequence1'
+        ])
+        self.assertNotIn('helloWorld', result.output)
```
Comment (ebolyen) [inline q2cli/tests/test_tools.py]: You'll want to check that it isn't in `output_path` rather than the stdout.

File: q2cli/tests/test_tools.py [hunk 5]
```diff
@@ -130,6 +135,64 @@ def test_merged_metadata(self):
...
+
```
Comment (ebolyen) [inline q2cli/tests/test_tools.py]: I was going to comment that we're missing a test to see if a directory is replaced, but in retrospect, that probably doesn't make a lot of sense. No reason to delete the directory before-hand, that could cause a lot of unhappy users.

File: q2cli/tools.py [hunk 6]
```diff
@@ -19,26 +19,35 @@ def tools():
...
+@q2cli.option('--input-path', required=True,
+              type=click.Path(exists=True, file_okay=True,
+                              dir_okay=False, readable=True))
+@q2cli.option('--output-path', required=True,
+              type=click.Path(exists=False, file_okay=True, dir_okay=True,
                               writable=True),
               help='Directory where data should be exported to')
-def export_data(path, output_dir):
+@q2cli.option('--output-format', required=False,
+              help='Format which the data should be exported as')
```
Comment (ebolyen) [inline q2cli/tools.py]: Mention that this cannot be used with a visualization.

File: q2cli/tools.py [hunk 7]
```diff
@@ -19,26 +19,35 @@ def tools():
...
-                    "Visualization and will support exporting to multiple "
-                    "formats in the future. For now, the data is exported in "
-                    "the format it is stored within the Artifact or "
-                    "Visualization. Use 'qiime tools extract' to extract the "
-                    "Artifact or Visualization's entire archive.")
-@click.argument('path', type=click.Path(exists=True, file_okay=True,
-                                        dir_okay=False, readable=True))
-@q2cli.option('--output-dir', required=True,
-              type=click.Path(exists=False, file_okay=False, dir_okay=True,
+               short_help='Export data from a QIIME 2 Artifact',
```
Comment (ebolyen) [inline q2cli/tools.py]: append: `or Visualization.`.

I think that got lost in the reflow

File: q2cli/tools.py [hunk 8]
```diff
@@ -19,26 +19,35 @@ def tools():
...
+               help="Exporting extracts the data stored in an Artifact"
```
Comment (ebolyen) [inline q2cli/tools.py]: Try:
`"Exporting extracts (and optionally transforms) data stored inside an Artifact or Visualization. Note that Visualizations cannot be transformed with --output-format."` reflow as needed for PEP8.

**Comment by ebolyen** [comment]
Looks like one of the tests is failing as well.

File: q2cli/tools.py (hunk 8 above)
Comment (turanoo) [reply q2cli/tools.py]: @ebolyen, thanks for reviewing the PR, I will go ahead and make these changes!

**Comment by turanoo** [comment]
@ebolyen I will have a look at that!

File: q2cli/tests/test_tools.py [hunk 9]
```diff
@@ -130,6 +135,70 @@ def test_merged_metadata(self):
...
+    def test_export_path_file_is_replaced(self):
+        output_path = os.path.join(self.tempdir, 'helloWorld')
+        os.mkdir(output_path)
+        self.runner.invoke(tools, [
+            'export', '--input-path', self.ints1, '--output-path', output_path,
+            '--output-format', 'IntSequence1'
+        ])
+
+        self.assertNotIn('helloWorld', os.listdir(output_path))
```
Comment (ebolyen) [inline q2cli/tests/test_tools.py]: This isn't the right kind of assertion, remember, we're looking to see that the file is replaced.
The rough idea here is right, but we need to look inside a file not a directory.

Comment (turanoo) [reply q2cli/tests/test_tools.py]: Yeah, you're right! Thanks for the reminding me.

**Comment by ebolyen** [comment]
@thermokarst, what are you thoughts on merging this so close to the release? We'll probably run into documentation updates since we're breaking the API. It might also be nice to break the API on `import` to match (`--source-format` -> `--input-format`).

**Comment by ebolyen** [comment]
Discussed offline, queued for next release cycle :factory:

**Pull request merged**